components/datetime/src/external_loaders.rs

```diff
@@ -0,0 +1,161 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! Internal traits and structs for loading data from other crates.
+
+use icu_calendar::provider::WeekDataV2Marker;
+use icu_calendar::week::WeekCalculator;
+use icu_calendar::CalendarError;
+use icu_decimal::options::FixedDecimalFormatterOptions;
+use icu_decimal::provider::DecimalSymbolsV1Marker;
+use icu_decimal::{DecimalError, FixedDecimalFormatter};
+use icu_provider::prelude::*;
+
+/// Trait for loading a FixedDecimalFormatter.
+///
+/// Implemented on the provider-specific loader types in this module.
+pub(crate) trait FixedDecimalFormatterLoader {
```

Comment:
observation: must stay private, we may need to be able to load RBNF in the long run

in the medium term we will probably hardcode a `FixedDecimalFormatter | CustomFormat` where the CustomFormat is a simple enum with code attached.